examples/plot_text.py

```diff
@@ -0,0 +1,165 @@
+"""DiCoDiLe on text images
+==============================
+
+This example illlustrates pattern recovery on a noisy text image using
+DiCoDiLe algorithm.
+
+"""  # noqa
+import matplotlib.pyplot as plt
+import numpy as np
+
+from dicodile import dicodile
+from dicodile.data.images import fetch_pami
+from dicodile.update_d.update_d import tukey_window
+from dicodile.utils import check_random_state
+from dicodile.utils.csc import reconstruct
+from dicodile.utils.dictionary import init_dictionary, prox_d
+from dicodile.utils.viz import display_dictionaries
+
+
+###############################################################################
+# We will first load PAMI image generated from a text of **5000**
+# characters drawn uniformly from the **4** letters **P** **A** **M**
+# **I** and 2 whitespaces and assign it to `X`.
+#
+# We will also load the images of the four characters used to generate
+# `X` and assign it to variable `D`.
+
+X_original, D = fetch_pami()
+
+
+###############################################################################
+# We will work on the copy `X` of the original image and we need to
+# reshape image data `X` and `D` to fit to expected signal shape of
+# `dicodile`:
+#
+# (n_channels, *sig_support)
+
+X = X_original.copy()
+
+X = X.reshape(1, *X.shape)
+D = D.reshape(4, 1, *D.shape[-2:])
+
+# pad `D`
+D = np.pad(D, [(0, 0), (0, 0), (4, 4), (4, 4)])
+
+
+###############################################################################
+# Let's display an extract of the original text image `X_original` and
+# all the images of characters from `D`.
+
+extract_x = X_original[190:490, 250:750]
+plt.axis('off')
+plt.imshow(extract_x, cmap='gray')
+
+display_dictionaries(D)
+
+###############################################################################
+# We add a Gaussian white noise with standard deviation σ std (X) and σ = 3
+# to `X`.
+
+std = 3
+rng = check_random_state(None)
+
+X += std * X.std() * rng.randn(*X.shape)
+
+###############################################################################
+# We will create a random dictionary of **K = 10** patches from the
+# noisy image.
+
+# set number of patches
+n_atoms = 10
+# set individual atom (patch) size
+atom_support = np.array(D.shape[-2:])
+
+D_init = init_dictionary(X, n_atoms=n_atoms, atom_support=atom_support,
+                         random_state=rng)
+
+# normalize the atoms
+D_init = prox_d(D_init)
```

Comment:
I would not expose prox_d here.

Let's just do it with numpy

D_init /= np.sqrt(np.sum(D_init, axis=..., keepdims=True))

Reply:
Can we write a normalization function in utils which could be applied to all dictionaries of atoms?
In Mandrill example we normalize D_init by  D_init = D_init / D_init.max().

Reply:
D_init = D_init / D_init.max()

seems weird to me. I'll let @tomMoral comment
I would use prox_d there but however I am not fan of exposing
publicly internal jargon like `prox_d`. Nobody will get what it does from function name

Reply:
The `D_init / D_init.max()` is mainly here for display purposes to have images between `0` and `1` and should only be used in `display_dictionaries`.

For this one, I would say that `init_dicitonary` should properly normalize the initial dictionary (with `prox_d` internally no?

Reply:
Ok. Yes, actually it does. I hadn't checked it, I just followed the code in `benchmarks/dicodile_text.py`. 

Then I'll remove this part.

Reply:
Done!